static/src/javascripts/lib/robust.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+/*
+    Swallows (and reports) exceptions. Designed to wrap around modules at the "bootstrap" level.
+    For example "comments throwing an exception should not stop auto refresh"
+ */
+
+import reportError from 'lib/report-error';
+
+function catchErrors(fn: Function): ?Error {
+    let error;
+
+    try {
+        fn();
+    } catch (e) {
+        error = e;
+    }
+
+    return error;
+}
+
+function log(name: string, error: Error, customReporter?: Function): void {
+    let reporter = reportError;
+
+    if (customReporter) {
+        reporter = customReporter;
+    }
+
+    if (window.console && window.console.warn) {
+        window.console.warn('Caught error.', error.stack);
+    }
+
+    reporter(error, { module: name }, false);
+}
+
+function catchErrorsAndLog(name: string, fn: Function, reporter?: Function) {
+    const error = catchErrors(fn);
+
+    if (error) {
+        log(name, error, reporter);
+    }
+}
+
+function catchErrorsAndLogAll(modules: Array<any>) {
+    modules.forEach(pair => catchErrorsAndLog(pair[0], pair[1]));
```

Comment:
i don't think we can say "you must destructure" but we can say "be as explicit for the next person as possible". in this case, it's hard for me see what this is a pair of? destructuring could make it clearer, so something like:
```js
modules.forEach(module => {
    const [whateverThisIs, andWhateverThisis] = module;
    catchErrorsAndLog(whateverThisIs, andWhateverThisis);
})
```
might make it clearer? realise that's not how it was in the original

is that what you meant @reg